File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/listener/SearchViewOnQueryTextListener.java
package com.github.android.lvrn.lvrnproject.view.listener;

import android.support.v7.widget.SearchView;

import io.reactivex.subjects.ReplaySubject;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public class SearchViewOnQueryTextListener implements SearchView.OnQueryTextListener {

    private ReplaySubject<String> mQueryReplaySubject;

    public SearchViewOnQueryTextListener(ReplaySubject<String> queryReplaySubject) {
        mQueryReplaySubject = queryReplaySubject;
    }

    @Override
    public boolean onQueryTextSubmit(String query) {
        return true;
    }

    @Override
    public boolean onQueryTextChange(String newText) {
        mQueryReplaySubject.onNext(newText);
        return true;
    }
}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/listener/SearchViewOnQueryTextListener.kt
package com.github.android.lvrn.lvrnproject.view.listener

import androidx.appcompat.widget.SearchView
import io.reactivex.subjects.ReplaySubject

/**
 * @author Vadim Boitsov <vadimboitsov1></vadimboitsov1>@gmail.com>
 */
class SearchViewOnQueryTextListener(private val mQueryReplaySubject: ReplaySubject<String>) :
    SearchView.OnQueryTextListener {
    override fun onQueryTextSubmit(query: String): Boolean {
        return true
    }

    override fun onQueryTextChange(newText: String): Boolean {
        mQueryReplaySubject.onNext(newText)
        return true
    }
}
